source/extensions/filters/http/jwt_authn/jwks_cache.cc

```diff
@@ -41,10 +41,9 @@ class JwksDataImpl : public JwksCache::JwksData, public Logger::Loggable<Logger:
     }
     audiences_ = std::make_unique<::google::jwt_verify::CheckAudience>(audiences);
 
-    tls_.set([](Envoy::Event::Dispatcher&) { return std::make_shared<ThreadLocalCache>(); });
-
-    tls_->jwt_cache_ = JwtCache::create(jwt_provider.enable_jwt_cache(),
-                                        jwt_provider.jwt_cache_size(), time_source_);
+    tls_.set([this](Envoy::Event::Dispatcher&) {
+      return std::make_shared<ThreadLocalCache>(jwt_provider_, time_source_);
```

Comment:
don't pass `this` in the capture.   it will have race condition where config_filter is freed but lambda function is called
* pass (bool enable_jwt_cache, int jwt_cache_size) into the lambda capture.
* use the timeSource from the dispacher as

```
   bool enable_jwt_cache = provider.enable_jwt_cache();
   int jwt_cache_size = jwt_provider.jwt_cache_size();
   tls_.set([enable_jwt_cache, jwt_cache_size](Envoy::Event::Dispatcher& dispatcher) {
        std::make_shared<ThreadLocalCache>(enable_jwt_cache, jwt_cache_size, dispatcher.timeSource());
    });
```

Reply:
Done.

Reply:
PTAL